cmd/thor/node/node_benchmark_test.go

```diff
@@ -0,0 +1,374 @@
+package node
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"fmt"
+	"math"
+	"math/big"
+	"sync"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/cmd/thor/solo"
+	"github.com/vechain/thor/v2/genesis"
+	"github.com/vechain/thor/v2/logdb"
+	"github.com/vechain/thor/v2/muxdb"
+	"github.com/vechain/thor/v2/packer"
+	"github.com/vechain/thor/v2/state"
+	"github.com/vechain/thor/v2/test/datagen"
+	"github.com/vechain/thor/v2/test/testchain"
+	"github.com/vechain/thor/v2/thor"
+	"github.com/vechain/thor/v2/tx"
+)
+
+var (
+	cachedAccounts []genesis.DevAccount
+	once           sync.Once
+)
+
+func getCachedAccounts(b *testing.B) []genesis.DevAccount {
+	once.Do(func() {
+		cachedAccounts = createAccounts(b, 1_000)
+	})
+	return cachedAccounts
+}
+
+func BenchmarkBlockProcess_RandomSigners_ManyClausesPerTx(b *testing.B) {
```

Comment:
are these the same tests as the ones in the #892  ?

It might be better to merge that to master, then we can sync these branches with master

Reply:
I'd suggest we merge the other PR then and close this one. We can sync mainDB branch with master once the other is merged. Otherwise we get a merge nightmare

Reply:
Yes fair concern, I've been keeping tabs on the differences between them, so we should be good to go 👍